### PR TITLE
[NCL-7905] add transition times

### PR DESCRIPTION
**Pull request by michalovjan**
@thescouser89  The only caveat I have for my implementation is if I should have the Map<Transition, Instant> or Set<TransitionTimeDTO> in DTO layer. With the Map, I cannot sort it with comparator of Instant so that we have transitions lined up. With Set it's possible (and implemented).

### CODE REVIEW

File: dto/src/main/java/org/jboss/pnc/rex/dto/TaskDTO.java [hunk 1]
```diff
@@ -57,9 +61,11 @@ public class TaskDTO {
...
 
-    public Set<String> dependants = new HashSet<>();
+    public Set<String> dependants = new TreeSet<>();
 
-    public Set<String> dependencies = new HashSet<>();
+    public Set<String> dependencies = new TreeSet<>();
 
     public ConfigurationDTO configuration = new ConfigurationDTO();
+
+    private Map<Transition, Instant> timestamps = new TreeMap<>();
```
Comment (thescouser89) [inline dto/src/main/java/org/jboss/pnc/rex/dto/TaskDTO.java:70]: Why can't this also be a set of transitiontime or set of transitiontimedto?

**Comment by thescouser89** [comment]
@michalovjan I worry though that if you want to lock in the order of the timestamps in increasing order, that you shouldn't depend on a Set at the DTO level but as a List. With a set, it really depends on the implementation, whereas with a list, the order is guaranteed no matter the implementation (at the loss of duplicates that might creep in)

File: dto/src/main/java/org/jboss/pnc/rex/dto/TaskDTO.java (hunk 1 above)
Comment (michalovjan) [reply dto/src/main/java/org/jboss/pnc/rex/dto/TaskDTO.java:70]: It definitely can be. By accident, you've answered the question I've had at the start of PR :D. Lucky! Will do that and also make it a List as you've pointed out a very good question.

**Comment by thescouser89** [comment]
Internally using treeset, but dtos using List :+1: 

LGTM, and the transition time info is also very useful to have in general. Thanks for the PR Jan! :heart:

**Review by thescouser89 — APPROVED** [review]

**Pull request merged**